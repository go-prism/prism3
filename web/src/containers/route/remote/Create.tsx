import React, {useState} from "react";
import {
	Badge,
	Button,
	FormControlLabel,
	FormGroup,
	FormLabel,
	makeStyles,
	Radio,
	RadioGroup,
	Theme,
	Tooltip,
	Typography
} from "@material-ui/core";
import Icon from "@mdi/react";
import {mdiInformationOutline} from "@mdi/js";
import {useTheme} from "@material-ui/core/styles";
import {Link} from "react-router-dom";
import {Code, ValidatedData, ValidatedTextField} from "jmp-coreui";
import {Alert} from "@material-ui/lab";
import StandardLayout from "../../layout/StandardLayout";
import {DataIsValid} from "../../../utils/data";
import useLoading from "../../../hooks/useLoading";
import useErrors from "../../../hooks/useErrors";
import getErrorMessage from "../../../selectors/getErrorMessage";
import {REMOTE_ARCHETYPES} from "../../../config/constants";

const useStyles = makeStyles((theme: Theme) => ({
	title: {
		fontFamily: "Manrope",
		fontWeight: 500
	},
	form: {
		marginTop: theme.spacing(1)
	},
	formItem: {
		margin: theme.spacing(1)
	},
	formIcon: {
		paddingTop: theme.spacing(1.75)
	},
	flex: {
		display: "flex"
	},
	grow: {
		flexGrow: 1
	},
	button: {
		fontFamily: "Manrope",
		fontWeight: 600,
		textTransform: "none"
	}
}));

const initialURL: ValidatedData = {
	value: "",
	error: "",
	regex: new RegExp(/https?:\/\/(?:w{1,3}\.)?[^\s.]+(?:\.[a-z]+)*(?::\d+)?(?![^<]*(?:<\/\w+>|\/?>))/)
}

const initialName: ValidatedData = {
	value: "",
	error: "",
	regex: new RegExp(/^.{3,}$/)
}

const CreateRemote: React.FC = (): JSX.Element => {
	// hooks
	const classes = useStyles();
	const theme = useTheme();

	// global state
	const loading = useLoading([]);
	const error = useErrors([]);


	// local state
	const [arch, setArch] = useState<string>("generic");
	const [url, setURL] = useState<ValidatedData>(initialURL);
	const [name, setName] = useState<ValidatedData>(initialName);

	const handleArchChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
		setArch((e.target as HTMLInputElement).value);
	};

	const handleCreate = (): void => {
		// dispatch(createRemote({
		// 	allowlistList: [],
		// 	blocklistList: [],
		// 	archetype: arch,
		// 	name: name.value,
		// 	uri: url.value,
		// 	enabled: true,
		// 	striprestricted: false,
		// 	restrictedheadersList: []
		// })).then((action) => {
		// 	// only change the url if there was a success
		// 	if(action.error !== true)
		// 		history.push("/settings/remotes");
		// });
	}

	return (
		<StandardLayout>
			<div>
				<Typography
					className={classes.title}
					color="textPrimary"
					variant="h4">
					Create remote
				</Typography>
				<FormGroup
					className={classes.form}>
					<FormLabel
						className={classes.formItem}
						component="legend">
						General
					</FormLabel>
					<ValidatedTextField
						data={name}
						setData={setName}
						invalidLabel="Must be at least 3 characters."
						fieldProps={{
							className: classes.formItem,
							required: true,
							label: "Remote name",
							variant: "outlined",
							id: "txt-name"
						}}
					/>
					<ValidatedTextField
						data={url}
						setData={setURL}
						invalidLabel="Must be a valid URL."
						fieldProps={{
							className: classes.formItem,
							required: true,
							label: "Remote URL",
							variant: "outlined",
							id: "txt-url"
						}}
					/>
					<FormLabel
						className={classes.formItem}
						component="legend">
						Archetype
						<Tooltip title="The Archetype describes any special handling that this remote may require. E.g. not caching maven-metadata.xml">
							<Icon
								className={classes.formIcon}
								path={mdiInformationOutline}
								color={theme.palette.secondary.main}
								size={1}
							/>
						</Tooltip>
					</FormLabel>
					<Alert
						severity="info">
						Some archetypes are available for Early Access, however they are not production ready.
						They are indicated by the&emsp;<Badge badgeContent="EA" color="primary"/>.
					</Alert>
					<RadioGroup
						className={classes.formItem}
						aria-label="archetype"
						name="archetype"
						value={arch}
						onChange={handleArchChange}>
						{REMOTE_ARCHETYPES.map(a => <FormControlLabel
							key={a.name}
							control={<Radio
								color="primary"
							/>}
							label={a.stable ? <>
								{a.name}
							</> : <Badge
								badgeContent="EA"
								color="primary">
								{a.name}
							</Badge>}
							value={a.name.toLocaleLowerCase()}
						/>)}
					</RadioGroup>
					{error != null && <Alert
						severity="error">
						Failed to create Remote.
						<br/>
						<Code>
							{getErrorMessage(error)}
						</Code>
					</Alert>}
					<div
						className={`${classes.formItem} ${classes.flex}`}>
						<Button
							className={classes.button}
							component={Link}
							to="/settings/remotes"
							variant="outlined">
							Cancel
						</Button>
						<div className={classes.grow}/>
						<Button
							className={classes.button}
							style={{color: theme.palette.success.contrastText, backgroundColor: theme.palette.success.main}}
							disabled={!DataIsValid(url) || !DataIsValid(name) || loading}
							onClick={handleCreate}
							variant="contained">
							Create
						</Button>
					</div>
				</FormGroup>
			</div>
		</StandardLayout>
	);
}
export default CreateRemote;
