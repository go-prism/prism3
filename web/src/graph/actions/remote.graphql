mutation createRemote($name: String!, $uri: String!, $archetype: Archetype!, $transport: ID!) {
    createRemote(input: {name: $name, uri: $uri, archetype: $archetype, transport: $transport}) {
        id
    }
}
query getRemote($id: ID!) {
    getRemote(id: $id) {
        id
        createdAt
        updatedAt
        name
        uri
        archetype
        enabled
        transport {
            id
            name
            cert
            key
            ca
            skipTLSVerify
            httpProxy
            httpsProxy
            noProxy
        }
    }
}
query listRemotes($arch: String!) {
    listRemotes(arch: $arch) {
        id
        name
        updatedAt
        createdAt
        uri
        enabled
        archetype
    }
}