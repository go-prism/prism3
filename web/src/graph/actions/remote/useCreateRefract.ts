import {gql, MutationTuple, useMutation} from "@apollo/client";
import {Archetype, Refraction} from "../../types";

interface Data {
	createRefraction: Refraction;
}

interface Vars {
	name: string;
	archetype: Archetype;
	remotes: string[];
}

export const useCreateRefract = (): MutationTuple<Data, Vars> => {
	return useMutation(gql`
		mutation createRefract($name: String!, $archetype: Archetype!, $remotes: [ID!]!) {
			createRefraction(input: {name: $name, archetype: $archetype, remotes: $remotes}) {
				id
			}
		}
	`);
}
export default useCreateRefract;
