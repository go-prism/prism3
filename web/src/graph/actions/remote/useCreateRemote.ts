import {gql, MutationTuple, useMutation} from "@apollo/client";
import {Archetype, Remote} from "../../types";

interface Data {
	createRemote: Remote;
}

interface Vars {
	name: string;
	uri: string;
	archetype: Archetype;
	transport: string;
}

export const useCreateRemote = (): MutationTuple<Data, Vars> => {
	return useMutation(gql`
		mutation createRemote($name: String!, $uri: String!, $archetype: Archetype!, $transport: ID!) {
			createRemote(input: {name: $name, uri: $uri, archetype: $archetype, transport: $transport}) {
				id
			}
		}
	`);
}
export default useCreateRemote;
