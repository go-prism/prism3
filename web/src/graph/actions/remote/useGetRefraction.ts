import {gql, QueryTuple, useLazyQuery} from "@apollo/client";
import {Refraction} from "../../types";

interface Data {
	getRefraction: Refraction;
}

interface Vars {
	id: string;
}

export const useGetRefraction = (): QueryTuple<Data, Vars> => {
	return useLazyQuery(gql`
		query getRefraction($id: ID!) {
			getRefraction(id: $id) {
				id
				createdAt
				updatedAt
				name
				archetype
				remotes {
					id
					name
                }
			}
		}
	`);
}
export default useGetRefraction;
