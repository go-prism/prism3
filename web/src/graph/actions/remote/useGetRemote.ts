import {gql, QueryTuple, useLazyQuery} from "@apollo/client";
import {Remote} from "../../types";

interface Data {
	getRemote: Remote;
}

interface Vars {
	id: string;
}

export const useGetRemote = (): QueryTuple<Data, Vars> => {
	return useLazyQuery(gql`
		query getRemote($id: ID!) {
			getRemote(id: $id) {
				id
				createdAt
				updatedAt
				name
				uri
				archetype
				enabled
				transport {
					id
					name
					cert
					key
					ca
					skipTLSVerify
					httpProxy
					httpsProxy
					noProxy
                }
			}
		}
	`);
}
export default useGetRemote;
