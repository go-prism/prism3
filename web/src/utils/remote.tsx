import React, {ReactNode} from "react";
import Icon from "@mdi/react";
import {mdiServer} from "@mdi/js";
import {Theme} from "@material-ui/core";
import {Archetype} from "../graph/types";

export const getRemoteIcon = (theme: Theme, type: Archetype): ReactNode => {
	switch (type) {
		case Archetype.HELM:
		case Archetype.ALPINE:
		case Archetype.MAVEN:
		case Archetype.NPM:
		case Archetype.GO:
			return <img
				src={`/${type.toLocaleLowerCase()}_logo.svg`}
				alt={`${type} logo`}
				width={24}
				height={24}
			/>
		default:
			return <Icon
				path={mdiServer}
				color={theme.palette.text.secondary}
				size={1}
			/>
	}
}