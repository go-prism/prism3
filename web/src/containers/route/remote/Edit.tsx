import React, {useEffect, useMemo, useState} from "react";
import {
	Button,
	Chip,
	FormControlLabel,
	FormGroup,
	FormLabel,
	List,
	ListItem,
	ListItemIcon,
	ListItemText,
	makeStyles,
	Switch,
	Theme,
	Typography
} from "@material-ui/core";
import Icon from "@mdi/react";
import {mdiPencilOutline, mdiPlusCircleOutline, mdiShapeOutline} from "@mdi/js";
import {useTheme} from "@material-ui/core/styles";
import {Link, useHistory} from "react-router-dom";
import {Code, ValidatedData, ValidatedTextField} from "jmp-coreui";
import {Alert, Skeleton} from "@material-ui/lab";
import {UpdateProfileRequest} from "@prism/prism-rpc/build/gen/service/api/remote_pb";
import {useParams} from "react-router";
import {formatDistanceToNow} from "date-fns";
import StandardLayout from "../../layout/StandardLayout";
import {DataIsValid} from "../../../utils/data";
import getErrorMessage, {getGraphErrorMessage} from "../../../selectors/getErrorMessage";
import ExpandableListItem from "../../list/ExpandableListItem";
import {MetadataChip} from "../../../config/types";
import {IDParams} from "../settings";
import useGetRemote from "../../../graph/actions/remote/useGetRemote";
import {Archetype} from "../../../graph/types";
import {getRemoteIcon} from "../../../utils/remote";
import RestrictedHeaders from "./options/RestrictedHeaders";
import FirewallRules from "./options/FirewallRules";
import {CLIENT_PROFILE_DEFAULT} from "./options/ClientConfig";

const useStyles = makeStyles((theme: Theme) => ({
	title: {
		fontFamily: "Manrope",
		fontWeight: 500
	},
	form: {
		marginTop: theme.spacing(1)
	},
	formItem: {
		margin: theme.spacing(1)
	},
	formIcon: {
		paddingTop: theme.spacing(1.75)
	},
	flex: {
		display: "flex"
	},
	grow: {
		flexGrow: 1
	},
	button: {
		fontFamily: "Manrope",
		fontWeight: 600,
		textTransform: "none"
	},
	buttonDisabled: {
		backgroundColor: theme.palette.action.disabled,
		color: theme.palette.getContrastText(theme.palette.action.disabled)
	},
	chip: {
		margin: theme.spacing(0.5)
	}
}));

const initialURL: ValidatedData = {
	value: "",
	error: "",
	regex: new RegExp(/https?:\/\/(?:w{1,3}\.)?[^\s.]+(?:\.[a-z]+)*(?::\d+)?(?![^<]*(?:<\/\w+>|\/?>))/)
}

const initialName: ValidatedData = {
	value: "",
	error: "",
	regex: new RegExp(/^.{3,}$/)
}

const EditRemote: React.FC = (): JSX.Element => {
	// hooks
	const classes = useStyles();
	const theme = useTheme();
	const history = useHistory();
	const {id} = useParams<IDParams>();

	// global state
	const [getRemote, {data, loading, error}] = useGetRemote();

	// local state
	const [url, setURL] = useState<ValidatedData>(initialURL);
	const [name, setName] = useState<ValidatedData>(initialName);
	const [success, setSuccess] = useState<boolean>(false);
	const [enabled, setEnabled] = useState<boolean>(false);
	const [stripRestricted, setStripRestricted] = useState<boolean>(false);
	const [resHeaders, setResHeaders] = useState<string[]>([]);
	const [allowList, setAllowList] = useState<string[]>([]);
	const [blockList, setBlockList] = useState<string[]>([]);
	const [readOnly, setReadOnly] = useState<boolean>(false);

	const open = useMemo(() => {
		return history.location.hash.replace("#", "");
	}, [history.location.hash]);

	useEffect(() => {
		void getRemote({variables: {id: id}});
	}, [id]);

	const chips = useMemo(() => {
		const chipData: MetadataChip[] = [
			{
				label: data?.getRemote?.archetype || Archetype.NONE,
				icon: mdiShapeOutline
			},
			{
				label: formatDistanceToNow(new Date((data?.getRemote?.createdAt || 0) * 1000), {addSuffix: true}),
				icon: mdiPlusCircleOutline
			},
			{
				label: formatDistanceToNow(new Date((data?.getRemote?.updatedAt || 0) * 1000), {addSuffix: true}),
				icon: mdiPencilOutline
			},
		];
		return chipData.map(c => <Chip
			className={classes.chip}
			key={c.icon}
			label={c.label}
			icon={<Icon
				path={c.icon}
				size={0.75}
				color={theme.palette.text.secondary}
			/>}
			size="small"
		/>);
	}, [data?.getRemote]);

	useEffect(() => {
		if (data?.getRemote == null)
			return;
		setName({...name, value: data?.getRemote.name});
		setURL({...url, value: data?.getRemote.uri});
		setEnabled(data?.getRemote.enabled);
		// setResHeaders(data?.getRemote.security.authHeaders || []);
		// setAllowList(data?.getRemote.security.allowed || []);
		// setBlockList(data?.getRemote.security.blocked || []);
		setReadOnly(data?.getRemote.archetype === Archetype.GO);
	}, [data?.getRemote]);

	const hasChanged = (): boolean => {
		if (data?.getRemote == null)
			return false;
		if (name.value !== data?.getRemote.name)
			return true;
		if (url.value !== data?.getRemote.uri)
			return true;
		return enabled !== data?.getRemote.enabled;
	};

	const handleUpdate = (): void => {
		// if (remote == null) return;
		// setSuccess(false);
		// dispatch(updateRemote(id, {
		// 	allowlistList: allowList,
		// 	blocklistList: blockList,
		// 	archetype: remote.archetype,
		// 	name: name.value,
		// 	uri: url.value,
		// 	enabled: enabled,
		// 	striprestricted: stripRestricted,
		// 	restrictedheadersList: resHeaders,
		// 	clientprofile: profile
		// })).then((action) => {
		// 	// only show the alert if there was a success
		// 	if(action.error !== true) {
		// 		setSuccess(true);
		// 		// reload the remote
		// 		dispatch(getRemote(id));
		// 	}
		// });
	}

	const handleOpen = (id: string): void => {
		history.push({
			...history.location,
			hash: open === id ? "" : id
		});
	}

	const getOptions = () => {
		const data = [
			{
				id: "firewall-rules",
				primary: "Firewall rules",
				secondary: "Firewall rules define when this remote should be used or skipped.",
				children: <FirewallRules
					allowRules={allowList}
					blockRules={blockList}
					setAllowRules={setAllowList}
					setBlockRules={setBlockList}
					loading={loading}
					disabled={readOnly}
				/>
			},
			{
				id: "restricted-headers",
				primary: "Restricted headers",
				secondary: "Restricted headers control the behaviour of cache partitioning and remote authentication.",
				children: <RestrictedHeaders
					stripRestricted={stripRestricted}
					setStripRestricted={setStripRestricted}
					restrictedHeaders={resHeaders}
					setRestrictedHeaders={setResHeaders}
					loading={loading}
					disabled={readOnly}
				/>
			},
			{
				id: "transport",
				primary: "HTTP/TLS/Proxy options",
				secondary: "Configure how Prism communicates with remotes.",
				children: <div/>
			}
		];
		return data.map(d => <ExpandableListItem
			key={d.id}
			primary={d.primary}
			secondary={d.secondary}
			open={open === d.id}
			setOpen={o => {
				handleOpen(o ? d.id : "");
			}}>
			{d.children}
		</ExpandableListItem>);
	};

	return (
		<StandardLayout>
			<div>
				{error != null && <Alert
					severity="error">
					Failed to fetch Remote.
					<br/>
					<Code>
						{getGraphErrorMessage(error)}
					</Code>
				</Alert>}
				{success && <Alert
					severity="success">
					Remote updated successfully
				</Alert>}
				<ListItem>
					<ListItemIcon>
						{loading ? <Skeleton variant="circle" animation="wave" width={48} height={48}/> : getRemoteIcon(theme, data?.getRemote?.archetype || Archetype.NONE)}
					</ListItemIcon>
					<ListItemText
						disableTypography
						secondary={<Typography
							color="textSecondary">
							{loading ? <Skeleton animation="wave" width="15%"/> : `Remote ID: ${data?.getRemote?.id}`}
						</Typography>}>
						<Typography
							className={classes.title}
							color="textPrimary"
							variant="h4">
							{loading ? <Skeleton animation="wave" width="25%" height={64}/> : data?.getRemote?.name}
						</Typography>
					</ListItemText>
				</ListItem>
				<FormGroup
					className={classes.form}>
					<div>
						{loading ? <Skeleton
							animation="wave"
							width="35%"
							height={32}
						/> : chips}
					</div>
					<FormLabel
						className={classes.formItem}
						component="legend">
						General
					</FormLabel>
					{!loading && readOnly && <Alert
						severity="warning">
						This Remote is read-only and cannot be modified.
					</Alert>}
					<ValidatedTextField
						data={name}
						setData={setName}
						invalidLabel="Must be at least 3 characters."
						fieldProps={{
							className: classes.formItem,
							required: true,
							label: "Remote name",
							variant: "outlined",
							id: "txt-name",
							disabled: loading || readOnly
						}}
					/>
					<ValidatedTextField
						data={url}
						setData={setURL}
						invalidLabel="Must be a valid URL."
						fieldProps={{
							className: classes.formItem,
							required: true,
							label: "Remote URL",
							variant: "outlined",
							id: "txt-url",
							disabled: loading || readOnly
						}}
					/>
					{/*<Collapse*/}
					{/*	in={remote != null && !enabled && enabled !== remote?.enabled}>*/}
					{/*	<Alert*/}
					{/*		className={classes.formItem}*/}
					{/*		severity="warning">*/}
					{/*		Disabling a remote stops new data from being fetched from it.*/}
					{/*		Data that has already been downloaded and cached will still be available.*/}
					{/*	</Alert>*/}
					{/*</Collapse>*/}
					<FormControlLabel
						className={classes.formItem}
						control={<Switch
							color="primary"
							checked={enabled}
							disabled={readOnly}
							onChange={(_, checked) => setEnabled(checked)}
						/>}
						label="Enabled"
					/>
					<List>
						{getOptions()}
					</List>
					{error != null && <Alert
						severity="error">
						Failed to update Remote.
						<br/>
						<Code>
							{getErrorMessage(error)}
						</Code>
					</Alert>}
					<div
						className={`${classes.formItem} ${classes.flex}`}>
						<Button
							className={classes.button}
							component={Link}
							to="/settings/remotes"
							variant="outlined">
							Cancel
						</Button>
						<div className={classes.grow}/>
						<Button
							className={classes.button}
							color="primary"
							classes={{
								disabled: classes.buttonDisabled
							}}
							disabled={!DataIsValid(url) || !DataIsValid(name) || loading || !hasChanged() || readOnly}
							onClick={handleUpdate}
							variant="contained">
							Save changes
						</Button>
					</div>
				</FormGroup>
			</div>
		</StandardLayout>
	);
}
export default EditRemote;
