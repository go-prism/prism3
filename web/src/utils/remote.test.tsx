import renderer from "react-test-renderer";
import {createMuiTheme} from "@material-ui/core";
import React from "react";
import {getRemoteIcon} from "./remote";

describe("getRemoteIcon", () => {
	it("renders maven correctly", () => {
		const theme = createMuiTheme();
		const tree = renderer.create(<div>
			{getRemoteIcon(theme, "maven")}
		</div>).toJSON();
		expect(tree).toMatchSnapshot();
	});
	it("renders node correctly", () => {
		const theme = createMuiTheme();
		const tree = renderer.create(<div>
			{getRemoteIcon(theme, "node")}
		</div>).toJSON();
		expect(tree).toMatchSnapshot();
	});
	it("renders alpine correctly", () => {
		const theme = createMuiTheme();
		const tree = renderer.create(<div>
			{getRemoteIcon(theme, "alpine")}
		</div>).toJSON();
		expect(tree).toMatchSnapshot();
	});
	it("renders generic correctly", () => {
		const theme = createMuiTheme();
		const tree = renderer.create(<div>
			{getRemoteIcon(theme, "")}
		</div>).toJSON();
		expect(tree).toMatchSnapshot();
	});
})