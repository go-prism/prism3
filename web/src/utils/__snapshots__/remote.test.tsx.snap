// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getRemoteIcon renders alpine correctly 1`] = `
<div>
  <img
    alt="alpine logo"
    height={24}
    src="/alpine_logo.svg"
    width={24}
  />
</div>
`;

exports[`getRemoteIcon renders generic correctly 1`] = `
<div>
  <svg
    role="presentation"
    style={
      Object {
        "height": "1.5rem",
        "width": "1.5rem",
      }
    }
    viewBox="0 0 24 24"
  >
    <path
      d="M4,1H20A1,1 0 0,1 21,2V6A1,1 0 0,1 20,7H4A1,1 0 0,1 3,6V2A1,1 0 0,1 4,1M4,9H20A1,1 0 0,1 21,10V14A1,1 0 0,1 20,15H4A1,1 0 0,1 3,14V10A1,1 0 0,1 4,9M4,17H20A1,1 0 0,1 21,18V22A1,1 0 0,1 20,23H4A1,1 0 0,1 3,22V18A1,1 0 0,1 4,17M9,5H10V3H9V5M9,13H10V11H9V13M9,21H10V19H9V21M5,3V5H7V3H5M5,11V13H7V11H5M5,19V21H7V19H5Z"
      style={
        Object {
          "fill": "rgba(0, 0, 0, 0.54)",
        }
      }
    />
  </svg>
</div>
`;

exports[`getRemoteIcon renders maven correctly 1`] = `
<div>
  <img
    alt="maven logo"
    height={24}
    src="/maven_logo.svg"
    width={24}
  />
</div>
`;

exports[`getRemoteIcon renders node correctly 1`] = `
<div>
  <img
    alt="node logo"
    height={24}
    src="/node_logo.svg"
    width={24}
  />
</div>
`;
